File: server/src/main/java/com/fluffy/exam/domain/ExamRepositoryCustom.java
package com.fluffy.exam.domain;

import com.fluffy.exam.domain.dto.ExamSummaryDto;
import com.fluffy.exam.domain.dto.MyExamSummaryDto;
import com.fluffy.exam.domain.dto.SubmittedExamSummaryDto;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ExamRepositoryCustom {

    Page<ExamSummaryDto> findPublishedExamSummaries(Pageable pageable);

    Page<MyExamSummaryDto> findMyExamSummaries(Pageable pageable, ExamStatus status, Long memberId);

    Page<SubmittedExamSummaryDto> findSubmittedExamSummaries(Pageable pageable, Long memberId);
}

File: server/src/main/kotlin/com/fluffy/exam/domain/ExamRepositoryCustom.kt
package com.fluffy.exam.domain

import com.fluffy.exam.domain.dto.ExamDetailSummaryDto
import com.fluffy.exam.domain.dto.ExamSummaryDto
import com.fluffy.exam.domain.dto.MyExamSummaryDto
import com.fluffy.exam.domain.dto.SubmittedExamSummaryDto
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable

interface ExamRepositoryCustom {

    fun findPublishedExamSummaries(pageable: Pageable): Page<ExamSummaryDto>

    fun findMyExamSummaries(pageable: Pageable, status: ExamStatus, memberId: Long): Page<MyExamSummaryDto>

    fun findSubmittedExamSummaries(pageable: Pageable, memberId: Long): Page<SubmittedExamSummaryDto>

    fun findExamDetailSummary(examId: Long): ExamDetailSummaryDto?
}